tensorboard/uploader/proto/write_service.proto

```diff
@@ -287,15 +289,47 @@ message GetOrCreateBlobSequenceResponse {
   string blob_sequence_id = 1;
 }
 
+message GetBlobMetadataRequest {
+  // The ID of the BlobSequence of which this blob is a member.
+  string blob_sequence_id = 1;
+  // The position of this Blob within the BlobSequence
+  int64 index = 2;
+}
+
+message GetBlobMetadataResponse {
+  enum BlobState {
+    // Object state is unknown. This value should never be used; it is present
+    // only as a proto3 best practice.
+    // See https://developers.google.com/protocol-buffers/docs/proto3#enum
+    UNKNOWN = 0;
+
+    // Object is being written and not yet finalized.
+    UNFINALIZED = 1;
+
+    // Object is finalized.
+    CURRENT = 2;
+  }
+
+  // State of the object (still appending vs. complete).
+  BlobState object_state = 1;
+  // Size of the object in bytes.  This is the sum of the chunk sizes
+  // received from the stream so far.  In the response to the final chunk,
```

Comment:
This description is a bit stale - in the new context we don't necessarily have a stream or chunks, since we might send this request before even initiating a `WriteBlob` stream, and this also isn't ever in response to the final chunk.

Reply:
Good catch